scipy/_lib/tests/test_linear_assignment.py

```diff
@@ -0,0 +1,98 @@
+from itertools import product
+
+from numpy.testing import assert_array_equal
+import numpy as np
+
+from scipy.optimize import linear_sum_assignment
+from scipy.sparse import csr_matrix, random
+from scipy.sparse.csgraph import min_weight_full_bipartite_matching
+
+
+# Tests that combine scipy.optimize.linear_sum_assignment and
+# scipy.sparse.csgraph.min_weight_full_bipartite_matching
+def test_two_methods_give_expected_result_on_small_inputs():
+    solvers = [(linear_sum_assignment, np.array),
+               (min_weight_full_bipartite_matching, csr_matrix)]
+    signs = [-1, 1]
+    test_cases = [
+        # Square
+        ([[400, 150, 400],
+          [400, 450, 600],
+          [300, 225, 300]],
+         [150, 400, 300]
+         ),
+
+        # Rectangular variant
+        ([[400, 150, 400, 1],
+          [400, 450, 600, 2],
+          [300, 225, 300, 3]],
+         [150, 2, 300]),
+
+        ([[10, 10, 8],
+          [9, 8, 1],
+          [9, 7, 4]],
+         [10, 1, 7]),
+
+        # Square
+        ([[10, 10, 8, 11],
+          [9, 8, 1, 1],
+          [9, 7, 4, 10]],
+         [10, 1, 4]),
+
+        # Rectangular variant
+        ([[10, float("inf"), float("inf")],
+          [float("inf"), float("inf"), 1],
+          [float("inf"), 7, float("inf")]],
+         [10, 1, 7]),
+    ]
+    for (solver, array_type), sign, (cost_matrix, expected_cost)\
```

Comment:
minor: we usually do line continuations with parentheses instead of `\`, though not sure that's worth flushing the CI again for

Reply:
Changed to

```python
    for (
            (solver, array_type), sign, (cost_matrix, expected_cost)
    ) in product(solvers, signs, test_cases):
```

I'll make sure to bundle all changes in one push. But it's interesting to see that you value CI time over minor code improvements!